app/assets/javascripts/discourse/app/components/discovery-topics-list.js

```diff
@@ -21,9 +21,11 @@ const DiscoveryTopicsListComponent = Component.extend(UrlRefresh, LoadMore, {
     }
   },
 
-  @observes("topicTrackingState.states")
-  _updateTopics() {
-    this.topicTrackingState.updateTopics(this.model.topics);
+  @on("init")
+  _initialize() {
+    this.topicTrackingState.onStateChange(() =>
```

Comment:
This is a bit fishy to me. Every time the `discourse-topics-list` component is instantiated it's going to wire up this event?  If you visit a topics list page, leave it, come back, leave it, come back I feel like it would be leaking.

I recommend moving this to `didInsertElement` and having a corresponding method in `willDestroyElement` that removes the `onStateChange` event.

Reply:
Good call! I guess the observer was doing that transparently before? Will make that change.